src/main/groovy/de/dkfz/roddy/bam2fastq/BamToFastqWorkflow.groovy

```diff
@@ -162,12 +179,12 @@ class BamToFastqWorkflow extends Workflow {
         for (BaseFile bamFile : bamFiles) {
 
             if (config.outputPerReadGroup) {
-                ReadGroupGroup groups = readGroupsPerBamfile[bamFile.absolutePath]
+                ReadGroupGroup<BaseFile> groups = readGroupsPerBamfile[bamFile.absolutePath]
```

Comment:
If you define ReadGroupGroup with BaseFile like <? extends BaseFile>, you can omit it here.